lib/steps/step_list.go

```diff
@@ -48,3 +55,34 @@ func (stepList *StepList) Prepend(step Step) {
 func (stepList *StepList) PrependList(otherList StepList) {
 	stepList.List = append(otherList.List, stepList.List...)
 }
+
+// WrapOptions represents the options given to Wrap.
+type WrapOptions struct {
+	RunInGitRoot     bool
+	StashOpenChanges bool
+}
+
+// Wrap wraps the given StepList in steps that
+// change to the Git root directory or stash away open changes.
+func (stepList *StepList) Wrap(options WrapOptions) {
```

Comment:
We do always use `Wrap` in the commands but since we also user `StepList` elsewhere, like in the run state, where we don't wrap I prefer the logic as is.